flocker/control/test/test_script.py

```diff
@@ -75,19 +77,30 @@ class ControlScriptEffectsTests(SynchronousTestCase):
     """
     Tests for effects ``ControlScript``.
     """
+    def setUp(self):
+        """
+        Create some certificates to use when creating the control service.
+        """
+        ca_set, _ = get_credential_sets()
+        self.certificate_path = FilePath(self.mktemp())
+        self.certificate_path.makedirs()
+        ca_set.copy_to(self.certificate_path, control=True)
+
     def test_starts_http_api_server(self):
         """
         ``ControlScript.main`` starts a HTTP server on the given port.
         """
         options = ControlOptions()
-        options.parseOptions(
-            [b"--port", b"tcp:8001", b"--data-path", self.mktemp()])
+        options.parseOptions([
+            b"--port", b"tcp:8001", b"--data-path", self.mktemp(),
+            b"--certificates-directory", self.certificate_path.path
+        ])
         reactor = MemoryCoreReactor()
         ControlScript().main(reactor, options)
         server = reactor.tcpServers[0]
         port = server[0]
         factory = server[1].__class__
-        self.assertEqual((port, factory), (8001, Site))
+        self.assertEqual((port, factory), (8001, TLSMemoryBIOFactory))
```

Comment:
Also no longer really demonstrates that we have an HTTP server that serves up our API (though to be fair it didn't demonstrate anything about our API server before this change, only that it was HTTP-ish).